pkgs/dart_mcp/test/client_and_server_test.dart

```diff
@@ -287,6 +288,45 @@ void main() {
       );
       expect(environment.initializeServer(), completes);
     });
+
+    test('server exits before initialization', () {
+      final client = TestMCPClient();
+      final clientController = StreamController<String>();
+      final serverController = StreamController<String>();
+      final clientChannel = StreamChannel<String>.withGuarantees(
+        clientController.stream,
+        serverController.sink,
+      );
+      final serverChannel = StreamChannel<String>.withGuarantees(
+        serverController.stream,
+        clientController.sink,
+      );
+      final connection = client.connectServer(clientChannel);
+
+      expect(
+        connection.initialize(
+          InitializeRequest(
+            protocolVersion: ProtocolVersion.latestSupported,
+            capabilities: ClientCapabilities(),
+            clientInfo: ClientImplementation(name: '', version: ''),
+          ),
+        ),
+        throwsA(
+          isA<StateError>().having(
+            (e) => e.message,
+            'message',
+            'The client closed with pending request "initialize".',
+          ),
+        ),
+      );
+
+      serverChannel.sink.close();
```

Comment:
does this need to be before the initialize request? Maybe add a comment to the line of code that is closing the client